### PR TITLE
Add feature to exit with semihosting::process:abort()

### PR DESCRIPTION
**Pull request by t-moe**
Useful to get defmt_test working on esp32 riscv. ( https://github.com/knurling-rs/defmt/pull/796)

### CODE REVIEW

File: build.rs [hunk 1]
```diff
@@ -42,4 +42,8 @@ fn main() {
             "The `print-jtag-serial` feature is only supported by the ESP32-C3, ESP32-C6, ESP32-S3 and ESP32-H2 chips"
         );
     }
+
+    if cfg!(feature = "semihosting") && cfg!(feature = "halt-cores") {
```
Comment (MabezDev) [inline build.rs:46]: Could we also bail out if we enable semihosting on an xtensa target?

Comment (t-moe) [reply build.rs:46]: The `semihosting` crate already bails out if you try to use it on a target which is not supported.
https://github.com/taiki-e/semihosting/blob/main/src/lib.rs#L223C1-L233

Would you like an additional check here anyways?

**Comment by bjoernQ** [comment]
Just a thought: could we avoid the additional dependency? It's just a glorified `ebreak` and we really just want one syscall 🤔 

https://github.com/riscv-software-src/riscv-semihosting/blob/HEAD/riscv-semihosting-spec.adoc

However, that dependency isn't a show-stopper here - but I think it would be easy to avoid it. I can also imagine adding it now and replacing it later

**Comment by t-moe** [comment]
> Just a thought: could we avoid the additional dependency? It's just a glorified `ebreak` and we really just want one syscall 🤔

I actually set the PR status back to draft, because I can also get defmt_test working without modifications on esp-backtrace and esp-println by using defmt-rtt and panic-probe. I'm still evaluating which road to pursue....

**Comment by bjoernQ** [comment]
I think when using RTT the better option is to use defmt-rtt / panic-probe 👍

**Comment by jessebraham** [comment]
> I think when using RTT the better option is to use defmt-rtt / panic-probe 👍

I agree, however unfortunately `panic-probe` is currently ARM-specific, and needs updating to support our chips.

**Comment by t-moe** [comment]
> unfortunately `panic-probe` is currently ARM-specific, and needs updating to support our chips.

True. 

I'm working on that in https://github.com/knurling-rs/defmt/pull/796 , but waiting on feedback for now.

**Comment by bjoernQ** [comment]
We probably can close this? Feel free to re-open as needed

**Pull request closed without merging**